app/models/Backend.scala

```diff
@@ -783,6 +808,29 @@ class Backend @Inject() (implicit
     }
   }
 
+  def filterLiteratureByDate(
+                              pub: Publication,
+                              date: Option[Long],
+                              comparator: Option[ComparatorEnum.Comparator]) : Boolean =
+    {
+      if(date.isEmpty) true
+      else {
+        val format = new java.text.SimpleDateFormat("yyyy-MM-dd")
+        val d1: Long = format.parse(pub.date).getTime
```

Comment:
The filter here doesn't work I don't think. `getTime` returns the time in milliseconds since 1 Jan 1970 so we're going to have very large numbers. Date as provided through by the filter is going to be a year, say 2020. Unless we convert that into Millis first we will get the wrong results. I might be wrong if you are planning on having the FE pass in the date to the API in Millis?

I've checked in the DB and all the records start with YYYY, so you can always do something like `pub.date.take(4)` and use that to compare with the `date` provided. For security you can add in an `assert` statement checking that we extracted a valid year.

Reply:
Add year and month to the query arguments